src/app/(post-login)/submission/program-table/CancerTypes.tsx

```diff
@@ -18,14 +18,17 @@
  */
 'use client';
 
-const CancerTypes = ({ types }: { types: string[] }) => (
+import { Maybe } from '@/__generated__/graphql';
+
+const CancerTypes = ({ types }: { types: Maybe<Maybe<string>[]> | undefined }) => (
```

Comment:
Update Types to match API generated files

Reply:
Would it be better to `pick` the sub type of `cancerTypes` from here: https://github.com/icgc-argo/rdpc-ui/blob/develop/src/__generated__/graphql.ts#L2744

`Maybe` is just a utility type being imported from the generated types - so we may as well use the full generated type?

Reply:
In principal yes but in practice `Pick<ProgramsListQuery['programs'], 'cancerTypes'>` where both `programs` and `cancerTypes` are potentially undefined becomes tricky to untangle

Reply:
There's two options here:
1. Use the generated types from the GQL Schema:
```ts
import { Program } from '@/__generated__/graphql';

const CancerTypes = ({ types }: { types: Program['cancerTypes'] }) => (
```
This will ensure that your prop type can be read directly from a query on that generated type. The downside here is that if the GQL schema changes then you need to change this React Component.

2. Define the interface to accept only the type you want:

```ts
const CancerTypes = ({ types }: { types: string[] }) => (
```
and perform type filtering before passing data to your component:

```ts
			<CancerTypes
				types={
					cancerTypes
						? cancerTypes.filter((value): value is string => typeof value === 'string')
						: []
				}
			/>
```
This has the advantage of performing your type filtering in a location that is already aware that the data comes from GQL, and now your component never needs to care about where the data comes from: it is assured to get just the type it wants.

Reply:
Split the difference: 
config.tsx: `<CancerTypes types={cancerTypes || []} />`
CancerTypes.tsx: `const CancerTypes = ({ types }: { types: Program['cancerTypes'] | [] }) => (...)`

Reply:
Option 2 recommended. Keep the GQL knowledge limited to a few places, doesn't need to be in your display components.

Reply:
```ts
<CancerTypes types={cancerTypes || []} />
```
would be compatible with the type signature:
```ts
({ types }: { types: Program['cancerTypes']}); // no `| []` needed
```

The `| []` could cause you more problems. I wouldn't go that route.

Reply:
Would this work?
`<CancerTypes types={cancerTypes?.filter(notNull) || []} />`
`const CancerTypes = ({ types }: { types: string[] }) => (...)`

Reply:
Yeah that's basically a condensed form of what I wrote. I like it.